tests/failpoints/cases/test_merge.rs

```diff
@@ -1302,3 +1302,49 @@ fn test_merge_election_and_restart() {
     // If logs from different term are process correctly, store 2 should have latest updates.
     must_get_equal(&cluster.get_engine(2), b"k14", b"v14");
 }
+
+/// Testing that the source peer's read delegate should not be removed by the target peer
+/// and only removed when the peer is destroyed
+#[test]
+fn test_source_peer_read_delegate_after_apply() {
```

Comment:
The case describes in https://github.com/tikv/tikv/issues/11852 is hard to simulate in test, thus I test `ReadDelegate` won't be removed by target peer instead.